### PR TITLE
Add API for placeholder node creation.

### PR DESCRIPTION
**Pull request by noffle**
This PR has a few pieces to it:

1. Separates the osm-p2p inner module into two: DB creation, and the API. This
lets me substitute an in-memory osm-p2p-db to test the API without needing to
rely on React Native.

2. Generate a unique placeholder ID for OSM nodes created by the mobile app.
This ID also written to a special tag called 'osm-p2p-id' so that when the
coordinator app syncs with OSM.org, it will see that tag and dedupe the new
node with the original placeholder node.

3. Creates an `observation-link` document alongside the observation that
connects the node and the observation documents.

4. Unit tests that go over the different ways that observations on nodes can
exist (on a placeholder node, on an existing OSM.org node, on an OSM.org node
that used to be a placeholder node), to convince myself I understand how
that'll work.

### CODE REVIEW

File: app/screens/Observations/map.js [hunk 1]
```diff
@@ -54,7 +55,9 @@ class ObservationMapScreen extends Component {
     super();
 
     this.navigationOptions = { tabBarLabel: "Map" };
-    this._osm = osmp2p();
+    this._obsdb = createOsmp2p("obs");
+    this._osmdb = createOsmp2p("osm");
+    this._osm = osmp2p(this._obsdb, this._osmdb);
```
Comment (noffle) [inline app/screens/Observations/map.js]: Thanks for catching this!

**Comment by noffle** [comment]
Thanks for looking this over and handling the `master` merge @sethvincent! :100:

File: package.json [hunk 2]
```diff
@@ -10,7 +10,8 @@
     "ios": "react-native run-ios && react-native log-ios",
     "android": "react-native run-android && react-native log-android",
     "android:install": "react-native run-android && adb install android/app/build/outputs/apk/app-debug.apk && react-native log-android",
-    "precommit": "lint-staged"
+    "precommit": "lint-staged",
+    "test": "tap test/*.js"
```
Comment (sethvincent) [inline package.json]: should we change this to `tape test/*.js`?

Comment (noffle) [inline package.json]: Yes please! I was experimenting with `tape` vs `tap` and it looks like I didn't make my usage consistent across my PR

**Pull request merged**